src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,203 @@
+import io.sphere.sdk.retry.RetryPredicate;
+import io.sphere.sdk.retry.RetryRule;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClientConfig;
+
+import javax.annotation.Nonnull;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static java.lang.String.format;
+
+/**
+ * To create a Sphere Client with retry logic which computes a exponential backoff time delay in milliseconds.
+ * And handle all the configurations for the creation of client.
+ */
+public final class RetryableSphereClientWithExponentialBackoff {
+    protected static final long DEFAULT_MAX_DELAY = 60000;
+    protected static final long DEFAULT_INITIAL_RETRY_DELAY = 200;
+    protected static final int DEFAULT_MAX_RETRY_ATTEMPT = 5;
+    protected static final int DEFAULT_MAX_PARALLEL_REQUESTS = 20;
+    protected static final List<Integer> DEFAULT_STATUS_CODES_TO_RETRY = Arrays.asList(500, 502, 503, 504);
+
+    private final SphereClientConfig sphereClientConfig;
+    private long maxDelay;
+    private long initialRetryDelay;
+    private int maxRetryAttempt;
+    private int maxParallelRequests;
+    private List<Integer> statusCodesToRetry;
+
+    private RetryableSphereClientWithExponentialBackoff(@Nonnull final SphereClientConfig sphereClientConfig) {
+        this.sphereClientConfig = sphereClientConfig;
+        this.maxDelay = DEFAULT_MAX_DELAY;
+        this.initialRetryDelay = DEFAULT_INITIAL_RETRY_DELAY;
+        this.maxRetryAttempt = DEFAULT_MAX_RETRY_ATTEMPT;
+        this.maxParallelRequests = DEFAULT_MAX_PARALLEL_REQUESTS;
+        this.statusCodesToRetry = DEFAULT_STATUS_CODES_TO_RETRY;
+    }
+
+    /**
+     * Creates a new instance of {@link RetryableSphereClientWithExponentialBackoff} given a {@link SphereClientConfig}
+     * responsible for creation of a SphereClient.
+     *
+     * @param sphereClientConfig the client configuration for the client.
+     * @return the instantiated {@link RetryableSphereClientWithExponentialBackoff}.
+     */
+    public static RetryableSphereClientWithExponentialBackoff of(
+        @Nonnull final SphereClientConfig sphereClientConfig) {
+        return new RetryableSphereClientWithExponentialBackoff(sphereClientConfig);
+    }
+
+    /**
+     * Sets the maxDelay value value in milliseconds.
+     * @param maxDelay - build with maxDelay value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given maxDelay value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withMaxDelay(final long maxDelay) {
+        this.maxDelay = maxDelay;
+        return this;
+    }
+
+    /**
+     * Sets the initialDelay value in milliseconds.
+     * @param initialDelay - build with initialDelay value.
+     * If initialDelay is less than maxDelay then, a {@link IllegalArgumentException} will be thrown.
```

Comment:
If they're the same, will the error be thrown as well?

Reply:
Corrected, Comment was wrongly written. We throw error If initialDelay is equal or greater than maxDelay. https://github.com/commercetools/commercetools-sync-java/pull/623/commits/aa4f3630a0637338e522df40f926d29e005c995b